tests/components/tplink/test_light.py

```diff
@@ -534,3 +522,58 @@ def set_light_state_side_effect(state_data: dict):
     assert light_mock_data.get_sysinfo_mock.call_count < 40
     assert light_mock_data.get_light_state_mock.call_count < 40
     assert light_mock_data.set_light_state_mock.call_count < 10
+
+
+async def test_update_failure(
+    hass: HomeAssistant, light_mock_data: LightMockData, caplog
+):
+    """Test that update failures are logged."""
+
+    await async_setup_component(hass, HA_DOMAIN, {})
+    await hass.async_block_till_done()
+
+    await async_setup_component(
+        hass,
+        tplink.DOMAIN,
+        {
+            tplink.DOMAIN: {
+                CONF_DISCOVERY: False,
+                CONF_LIGHT: [{CONF_HOST: "123.123.123.123"}],
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    caplog.clear()
+    caplog.set_level(logging.WARNING)
+    await hass.helpers.entity_component.async_update_entity("light.light1")
+    assert caplog.text == ""
+
+    with patch("homeassistant.components.tplink.light.MAX_ATTEMPTS", 0):
+        caplog.clear()
+        caplog.set_level(logging.WARNING)
+        await hass.helpers.entity_component.async_update_entity("light.light1")
+        assert "Could not read state for 123.123.123.123|light1" in caplog.text
+
+    get_state_call_count = 0
+
+    def get_light_state_side_effect():
+        nonlocal get_state_call_count
+        get_state_call_count += 1
+
+        if get_state_call_count == 1:
+            raise SmartDeviceException()
+
+        return light_mock_data.light_state
+
+    light_mock_data.get_light_state_mock.side_effect = get_light_state_side_effect
+
+    with patch("homeassistant.components.tplink.light", MAX_ATTEMPTS=2, SLEEP_TIME=0):
+        caplog.clear()
+        caplog.set_level(logging.DEBUG)
+
+        await update_entity(hass, "light.light1")
+        assert (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)